### PR TITLE
Fix ProjectTemplate test failure logs on helix

### PR DESCRIPTION
**Pull request by BrennanConroy**
Binlogs aren't being uploaded on failure like expected. I believe this is because https://github.com/dotnet/aspnetcore/pull/37447 changed the directory ProjectTemplate tests write the binlog to, to artifacts/log, which likely isn't going to work correctly on Helix.

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Looks like the change is correct, binlog is now uploaded on failure: https://dev.azure.com/dnceng-public/public/_build/results?buildId=11140&view=ms.vss-test-web.build-test-results-tab&runId=219686&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab&resultId=100013

Sadly, it looks like the issue doesn't show up in the binlog.

**Comment by TanayParikh** [comment]
> Sadly, it looks like the issue doesn't show up in the binlog.

What's the next step here in that case?

**Comment by BrennanConroy** [comment]
> What's the next step here in that case?

Figure out the issue, this change is only about getting the binlogs uploaded to AzDo.
I hoped it would help solve the problem, but it didn't.

**Comment by TanayParikh** [comment]
> the issue

The issue would be the failures we're seeing here: https://github.com/dotnet/aspnetcore/pull/43932 ?

**Comment by BrennanConroy** [comment]
Yes

**Comment by brunolins16** [comment]
> > What's the next step here in that case?
> 
> Figure out the issue, this change is only about getting the binlogs uploaded to AzDo. I hoped it would help solve the problem, but it didn't.

I am investigating, without success ☹️, those errors but if I am getting it right, based on the output, usually the requested operation completed with sucess (eg.: Publish shows the output `publish` folder that happens after the publishing is completed), that is why the binlog does not reports anything, but even in this scenario the process ends with exit code `1`. Sounds like something wrong with the client itself.

File: src/ProjectTemplates/Shared/Project.cs [hunk 1]
```diff
@@ -30,14 +30,14 @@ public static string ArtifactsLogDir
...
+            var helixWorkItemUploadRoot = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
+            if (!string.IsNullOrEmpty(helixWorkItemUploadRoot))
             {
-                return testLogFolder;
+                return helixWorkItemUploadRoot;
             }
 
-            var helixWorkItemUploadRoot = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
-            return string.IsNullOrEmpty(helixWorkItemUploadRoot) ? GetAssemblyMetadata("ArtifactsLogDir") : helixWorkItemUploadRoot;
+            var testLogFolder = typeof(Project).Assembly.GetCustomAttribute<TestFrameworkFileLoggerAttribute>()?.BaseDirectory;
```
Comment (dougbu) [inline src/ProjectTemplates/Shared/Project.cs:39]: Nit: `TestFrameworkFileLoggerAttribute` should be added unconditionally and `BaseDirectory` should never be `null` based on a quick reading of Microsoft.AspNetCore.Testing.props. Might be time to at least remove the question mark here and `throw` if the attribute is missing for some strange reason.

**Review by dougbu — APPROVED** [review]

**Pull request merged**